### DISCUSSION TITLE
Failed to download Chrome for Win64

### DISCUSSION THREAD

**Discussion by fabercs** [hardkoded/puppeteer-sharp#2675] [Q&A]
PupeeterSharp Version: 18.0.2
The code below runs successfully on my localhost (.net 6 env), however when hosted to a shared hosting throws exception below;

**Failed to download Chrome for Win64 from https://storage.googleapis.com/chrome-for-testing-public/125.0.6422.76/win64/chrome-win64.zip**

The link is not broken, can download binaries as expected
Note: Due to group policy running exe on host is not allowed, I can not switch to deploy own chrome.exe option, at least what I am declared from hosting corp. (If there is another way to run via this option, it is also welcommed)

How can I solve this issue?

```
await new BrowserFetcher().DownloadAsync();
using var browser = await Puppeteer.LaunchAsync(new LaunchOptions
{
    Headless = true,
    Args = new []{ "--no-sandbox" }
 });
using var page = await browser.NewPageAsync();
```

**Comment by kblok** [comment]
@fabercs you can use a browser in the cloud service, such as https://www.browserless.io/

**Reply by fabercs** [reply]
@kblok thank you for your answer, so I wonder this is a known issue right? If hosting corp has policies/limitations on running exe on the server, it is completely inusable, am I right? (the services you suggest are expensive for my non-profit app right now)

**Reply by kblok** [reply]
If you can't run a browser, there is no browser to automate.
That's why we have a `ConnectAsync` option and companies like browserless.io offer those services. To help you out in these situations.
